src/runtime_src/driver/zynq/user/shim.cpp

```diff
@@ -151,7 +155,7 @@ size_t ZYNQShim::xclWrite(xclAddressSpace space, uint64_t offset, const void *ho
     return -1;
   }
 
-  if(XCL_ADDR_KERNEL_CTRL == space) {
+  if(XCL_ADDR_KERNEL_CTRL == space || XCL_ADDR_SPACE_DEVICE_PERFMON == space) {
```

Comment:
Hi @jvillarre. To support floating CU address, we changed the implementation of xclRead() and xclWrite(). This should be refine. I can work together with you to figure out how to do that.

Reply:
Just talk to Jason, let this pull request merge first. Then I can update my pull request.